feat: Appsmith helm with tls (#712)
appsmithorg/appsmith-docs: website/docs/getting-started/setup/instance-configuration/custom-domain/README.md

## Setting up SSL

SSL automatically set up for your instance once you configure a custom domain for your instance. All SSL certificates Appsmith creates are generated and kept up to date through [Let's Encrypt](https://letsencrypt.org).

## Custom SSL certificate

> The container supports to generate a free SSL certificate. If you have your owned certificate, please follow these steps to use it inside the container.

Author: ```suggestion
The container supports generating a free SSL certificate. If you have your own certificate, please follow these steps to use it inside the container.
```

* Firstly, please rename your certificate file as `fullchain.pem` and key file as `privkey.pem`
* Copy these files into the sub-directory `<mounting-directory>/ssl/` (_Note: Please change `<mounting-directory>` by the mounting volume directory in the `docker-compose.yml`. Default is `./stacks`_)
* Restart the container using `docker restart appsmith`

The container checks the certificate files in the folder `<mounting-directory>/ssl` and use them if they exist.

:::note

> In case the certificate files have a different name from `fullchain.pem` and `privkey.pem`, it considers missing custom certificate and auto provisioning the certificate by `Let's Encrypt`

Author: ```suggestion
In case the certificate files have a different name from `fullchain.pem` and `privkey.pem`, it will fail to find the  custom certificate and auto-provisions the certificate by `Let's Encrypt`
```

![](/img/custom\_domain.jpeg)

It redirects you to a new page. Add your domain name there. Once that's done, it'll give you records of the name servers. Copy the details of the NS (name servers). Use the custom name server's configuration on your domain provider. Sometimes, it might take up to 24 or 48 hours for this to go live.

:::tip
Your Appsmith instance should be available at [https://appsmith.mydomain.com](https://appsmith.mydomain.com) with automatic certificate provisioning and renewals.
:::

## Kubernetes

> The `APPSMITH_CUSTOM_DOMAIN` environment variable is no longer used for configuring TLS for a Kubernetes installation of Appsmith. To setup TLS on your Kubernetes installation of Appsmith, refer to the instructions [here](Configure the `APPSMITH_CUSTOM_DOMAIN field` in your [instance configuration](../) with your custom domain).

Reviewer: @sum35h - Where should the cross-reference to?

```
refer to the instructions [here](Configure the `APPSMITH_CUSTOM_DOMAIN field` in your [instance configuration](../) with your custom domain).
```

Should this section be added to Docker/K8s or should the `Docker/K8s` be renamed to Docker?

Author: That instruction should point to getting-started/setup/installation-guides/kubernetes#configure-tls

Author: Should this section be added to Docker/K8s or should the Docker/K8s be renamed to Docker? Sorry I did not understand this properly.

Reviewer: We already have a section on this page [Docker/K8s](https://docs.appsmith.com/getting-started/setup/instance-configuration/custom-domain#dockerk8s). Do we need to add the details for Kubernetes added in this PR to the above section or should we rename it to Docker and have a separate section for Kubernetes.
